noxfile.py

```diff
@@ -209,6 +221,13 @@ def unit(session):
     default(session)
 
 
+@nox.session(python=[UNIT_TEST_PYTHON_VERSIONS[0], UNIT_TEST_PYTHON_VERSIONS[-1]])
+def unit_noextras(session):
```

Comment:
Given that we run unit test on more than just 3.8 and 3.11, I think we can drop this session.

Reply:
@tswast 
The unit tests run with the extras installed.
This session is aimed at running the same tests when the extras are NOT installed.
On the unit testing level, do we want an `extras` installed vs `no extras` installed comparison?

Reply:
UNIT_TEST_EXTRAS_BY_PYTHON only has entries for Python 3.8 and 3.11. 

UNIT_TEST_PYTHON_VERSIONS has 3.7 through 3.11.

That means we run with no extras (besides those needed for tests) on 3.7, 3.9, and 3.10.

I figure that should be sufficient, especially given that UNIT_TEST_PYTHON_VERSIONS[0] is 3.7